storage/storage.c

```diff
@@ -469,16 +476,10 @@ static secbool is_not_wipe_code(const uint8_t *pin, size_t pin_len) {
 }
 
 static secbool ui_progress(uint32_t elapsed_ms) {
-  // elapsed_ms = number of elapsed milliseconds in the ongoing substep.
+  ui_rem -= elapsed_ms;
   if (ui_callback && ui_message) {
-    uint32_t progress = 0;
-    if (ui_total > 1000000) {
-      // Avoid overflow, precise enough.
-      progress = (ui_total - ui_rem) / (ui_total / 1000);
-    } else {
-      progress = ((ui_total - ui_rem) * 1000 + elapsed_ms) / ui_total;
-    }
```

Comment:
I am not sure why we were ever worried about overflow here, since the largest possible time in milliseconds is 2^16 * 1000 which fits well within uint32. We'd have to exceed 2^22 seconds for this to be a problem.

Reply:
It can overflow. I got lost in the change from measuring in seconds to milliseconds and the fact that progress is measured in units per mille. Fixed 5d7913436570052ef247a94c50f044964206d399.